core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -853,18 +853,24 @@ public Schema register(String subject,
   private void checkRegisterMode(
       String subject, Schema schema
   ) throws OperationNotPermittedException, SchemaRegistryStoreException {
+    String context = QualifiedSubject.qualifiedContextFor(tenant(), subject);
     if (isReadOnlyMode(subject)) {
-      throw new OperationNotPermittedException("Subject " + subject + " is in read-only mode");
+      throw new OperationNotPermittedException("Subject " + subject 
+      + " is in read-only mode, in context " 
+      + context);
```

Comment:
Phrasing it like this makes the message clearer
```suggestion
      throw new OperationNotPermittedException("Subject " + subject + " in context "
      + context + " is in read-only mode");
```
Also consider adding single or double quotations around the context name, because sometimes the context will just be `.` and that can also look confusing in an error message.
Example:
`Subject foo in context . is in read-only mode`
Example with the way you have it formatted currently (potentially more room for misinterpretation):
`Subject foo is in read-only mode, in context .`

Reply:
I don't think this would really be the case, since here I'm using qualifiedContextFor, so instead of just `in context .` it'll be `in context :.:` so I didn't think it would be much of an issue. I did make the changes for legibility though, thanks for pointing it out!